components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/service/DeviceOrganizationProviderServiceImpl.java

```diff
@@ -0,0 +1,331 @@
+    private List<DeviceOrganizationNode> transformMetadataHolderArray(List<DeviceOrganizationMetadataHolder> deviceDataList) {
+        List<DeviceOrganizationNode> nodeDataList = new ArrayList<>();
+        for (DeviceOrganizationMetadataHolder node : deviceDataList) {
+            nodeDataList.add(new DeviceOrganizationNode(node.getDeviceId(), node.getParent()));
+        }
+        return nodeDataList;
+    }
+
+    //this method takes arrays with node data and converts them into string arrays
+    private List<String> createChildIdList(List<DeviceOrganizationNode> deviceDataList) {
+        List<String> childIds = new ArrayList<>();
+        for (DeviceOrganizationNode holder : deviceDataList) {
+            childIds.add(holder.getId());
+        }
+        return childIds;
+    }
+
+    @Override
+    public String updateDeviceOrganizationName(String deviceId, String newDeviceName)
+            throws DeviceOrganizationException {
+        String updatedName = null;
+        try {
+            DeviceManagementDAOFactory.beginTransaction();
+            updatedName = deviceOrganizationDAOimpl.updateDeviceOrganizationName(deviceId, newDeviceName);
+            DeviceManagementDAOFactory.commitTransaction();
+        } catch (TransactionManagementException e) {
+            String msg = "Error occurred while initiating transaction";
+            log.error(msg, e);
+        } catch (DeviceOrganizationDAOException e) {
+            DeviceManagementDAOFactory.rollbackTransaction();
+            String msg = "Error while updating device name";
+            log.error(msg, e);
+            throw new DeviceOrganizationException(msg,e);
+        } finally {
+            DeviceManagementDAOFactory.closeConnection();
+            return updatedName;
+        }
+    }
+
+    @Override
+    public String updateDeviceOrganizationParent(String deviceId, String newParent) throws DeviceOrganizationException {
+        String updatedParent = null;
+        try {
+            DeviceManagementDAOFactory.beginTransaction();
+            updatedParent = deviceOrganizationDAOimpl.updateDeviceOrganizationParent(deviceId, newParent);
+            DeviceManagementDAOFactory.commitTransaction();
+        } catch (TransactionManagementException e) {
+            String msg = "Error occurred while initiating transaction";
+            log.error(msg, e);
+        } catch (DeviceOrganizationDAOException e) {
+            DeviceManagementDAOFactory.rollbackTransaction();
+            String msg = "Error occurred while updating device path";
+            log.error(msg, e);
+            throw new DeviceOrganizationException(msg,e);
+        } finally {
+            DeviceManagementDAOFactory.closeConnection();
+            return updatedParent;
+        }
+    }
+
+    @Override
+    public void removeDeviceOrganization(String deviceId) {
+        //not implemented
```

Comment:
if not yet implemented better throwing `UnsupportedOperationException`

Reply:
Fixed 49b7b1d9c3612c443fb0e827ce2d2e6b629022c9